src/xlsx.ts

```diff
@@ -118,53 +122,86 @@ function encodeCreativeWork(node: stencila.CreativeWork) {
 
 // Worksheet <-> Table
 
-function decodeTable(
-  name: string,
-  cells: { [key: string]: xlsx.CellObject }
-): stencila.Table {
-  return {
-    type: 'Table',
-    name,
-    rows: []
-    // TODO:
-    /*
-    cells: Object.entries(cells).map(function([key, cell]): stencila.TableCell {
-      let value = decodeCell(cell)
-      return {
-        type: 'TableCell',
-        name: key,
-        position: cellNameToPosition(key),
-        content: [value]
-      }
-    })
-    */
+interface Worksheet {
+  [cell: string]: xlsx.CellObject
+}
+
+interface RowMap {
+  [row: string]: {
+    [column: string]: stencila.TableCell
   }
 }
 
-function encodeTable(table: stencila.Table) {
-  const sheet: xlsx.WorkSheet = {}
-  // TODO:
-  // @ts-ignore
-  let cells = table.cells
-  if (cells) {
-    let maxCol = 0
-    let maxRow = 0
-    for (let cell of cells) {
-      if (!cell.position) continue
-      let [col, row] = cell.position
-      if (col > maxCol) maxCol = col
-      if (row > maxRow) maxRow = row
-      let name = cellPositionToName(cell.position)
-      let cellObject = encodeCell(cell.content[0] || null)
-      sheet[name] = cellObject
+/**
+ * Convert spreadsheet worksheet, a column-indexed table data format, to a RowMap,
+ * a row-indexed data representation.
+ *
+ * @param {xlsx.CellObject} cell
+ * @returns {stencila.TableCell}
+ */
+const worksheetToRowMap = (worksheet: Worksheet): RowMap =>
```

Comment:
@param and @returns don't look right. 

Although, I'm pretty obsessive - maybe too much so - about adding doc comments to functions. I have been wondering if it's necessary to add these tags since Typescript+TypeDoc does such a good job of auto-documenting them.